src/modules/jj.rs

```diff
@@ -0,0 +1,288 @@
+use super::{Context, Module, ModuleConfig};
+use crate::configs::jj::JJConfig;
+use crate::formatter::string_formatter::StringFormatterError;
+use crate::formatter::StringFormatter;
+use std::borrow::Cow;
+
+/// Creates a module with the jj status in the current directory
+///
+/// Will display the current branch, commit and change id of the jj repository in the current directory
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("jj_status");
+    let config = JJConfig::try_load(module.config);
+    // check if the current directory is a jj repo
+    context.exec_cmd("jj", &["root", "--ignore-working-copy"])?;
+
+    let template = jj_template(config.truncation_length);
+    let output = context.exec_cmd(
+        "jj",
+        &[
+            "log",
+            "-r",
+            "@",
+            "-T",
+            template.as_str(),
+            "--no-graph",
+            "--ignore-working-copy",
+        ],
+    )?;
```

Comment:
On my machine, when a little deep in a repo (not at the root), directly passing the root is faster by ~10ms.

Since it has already been computed in the `jj root` above, it could be passed in the command below:

```rust
let out = context.exec_cmd("jj", &["root", "--ignore-working-copy"])?.stdout;
let root = out.lines().next()?;

// and then
&["-R", &root, "log", ...]
```

Reply:
the issue is that it's annoying to mock the command for tests then

Reply:
done